antarest/matrixstore/matrix_garbage_collector.py

```diff
@@ -0,0 +1,119 @@
+            matrix_service.repo_dataset
+        )
+
+    def _get_saved_matrices(self) -> Set[str]:
+        logging.info("Getting all saved matrices")
+        return {f.split(".")[0] for f in listdir(self.saved_matrices_path)}
+
+    def _get_raw_studies_matrices(self) -> Set[str]:
+        logger.info("Getting all matrices used in raw studies")
+        return {
+            f.name.split(".")[0]
+            for f in self.managed_studies_path.rglob("*.link")
+        }
+
+    def _get_variant_studies_matrices(self) -> Set[str]:
+        logger.info("Getting all matrices used in variant studies")
+        command_blocks: List[CommandBlock] = db.session.query(
+            CommandBlock
+        ).all()
+        variant_study_commands = [
+            icommand
+            for c in command_blocks
+            for icommand in self.variant_study_service.command_factory.to_icommand(
+                c.to_dto()
+            )
+        ]
+        matrices = {
+            matrix
+            for command in variant_study_commands
+            for matrix in command.get_inner_matrices()
+        }
+        return matrices
+
+    def _get_datasets_matrices(self) -> Set[str]:
+        logger.info("Getting all matrices used in datasets")
+        datasets = self.dataset_repository.get_all_datasets()
+        return {
+            matrix.matrix_id
+            for dataset in datasets
+            for matrix in dataset.matrices
+        }
+
+    def _get_used_matrices(self) -> Set[str]:
+        """Return all matrices used in raw studies, variant studies and datasets"""
+        raw_studies_matrices = self._get_raw_studies_matrices()
+        variant_studies_matrices = self._get_variant_studies_matrices()
+        datasets_matrices = self._get_datasets_matrices()
+        return (
+            raw_studies_matrices | variant_studies_matrices | datasets_matrices
+        )
+
+    def _delete_unused_saved_matrices(self, unused_matrices: Set[str]) -> None:
+        """Delete all files with the name in unused_matrices"""
+        logging.info("Deleting unused saved matrices:")
+        for unused_matrix_id in unused_matrices:
+            logging.info(f"Deleting {unused_matrix_id}")
+            self.matrix_service.delete(unused_matrix_id)
+
+    def _clean_matrices(self) -> None:
+        """Delete all matrices that are not used anymore"""
+        stopwatch = StopWatch()
+        logger.info("Beginning of the cleaning process")
+        saved_matrices = self._get_saved_matrices()
+        used_matrices = self._get_used_matrices()
+        unused_matrices = saved_matrices - used_matrices
+        self._delete_unused_saved_matrices(unused_matrices=unused_matrices)
+        stopwatch.log_elapsed(
+            lambda x: logger.info(f"Finished cleaning matrices in {x}s")
+        )
+
+    def _loop(self) -> None:
+        while True:
+            self._clean_matrices()
+            logging.info("Sleeping for 1 hour")
+
+            time.sleep(3600)
```

Comment:
make this value configurable with a default to 3600

Reply:
you can put this in the storage config.
don't forget to add it in the resources/application.yml (for documentation purpose)